src/ansiblelint/yaml_utils.py

```diff
@@ -816,10 +818,12 @@ def __init__(
               tasks:
                 - name: Task
         """
-        # Default to reading/dumping YAML 1.1 (ruamel.yaml defaults to 1.2)
-        self._yaml_version_default: tuple[int, int] = (1, 1)
-        self._yaml_version: str | tuple[int, int] = self._yaml_version_default
-
+        if version:
+            if isinstance(version, str):
+                x, y = version.split(".", maxsplit=1)
+                version = (int(x), int(y))
+            self._yaml_version_default: VersionType = version
+            self._yaml_version: VersionType = self._yaml_version_default
```

Comment:
```suggestion
        if isinstance(version, str):
            x, y = version.split(".", maxsplit=1)
            version = (int(x), int(y))
        self._yaml_version_default: VersionType | None = version
        self._yaml_version: VersionType | None = self._yaml_version_default
        self._strip_yaml_version_directive = version is not None
```

Reply:
I do not want to touch this right now, especially as I know that a newer patch of the library introduces some data that we might want to use, look at 0.18.4 changelog entries https://pypi.org/project/ruamel.yaml/

We can do extra polishing in follow-up, but I prefer to use whatever they have instead of setting new properties that have meaning only for our subclass. I need to play with it bit more too see what data I have available.